architecture.tex

```diff
@@ -85,26 +87,152 @@
 downside to using Python directly instead of a custom DSL, and we consider it
 to be worth it for the advantages listed above.
```

Comment:
The introduction (which I will write) will consist of rationale for using Python.